README.md

```diff
@@ -1,2 +1,28 @@
-# miden-formatting
-Pretty-printing and formatting utilities for Miden crates
+# `miden-formatting`
+
+This crate provides some general infrastructure for pretty-printers and value foramtting that is needed by various Miden crates. Rather than implement this
+stuff in every place where it is needed, we've extracted the most important and general bits and put them in this crate.
+
+## Usage
+
+Add `miden-formatting` to your `Cargo.toml`:
+
+```toml
+[dependencies]
+miden-formatting = "0.1"
+```
+
+For `#![no_std]` builds:
+
+```toml
+[dependencies]
+miden-formatting = { version = "0.1", default-features = false }
+```
+
+There is a `std` feature you can use to conditionally enable functionality that requires libstd to implement. For now this features is not actually needed, but is likely to be used in the future, so we're providing it now.
+
+## Intro
+
+Most likely you are pulling in this crate to make use of the pretty-printer infrastructure. See the documentation for the `PrettyPrint` trait for a comprehensive intro to how to get started with it.
+
+You may also be interested in the example syntax tree defined [here](./formatting/src/prettier/tests.rs). This makes use of most features of the pretty printer in a small made-up language for learning.
```

Comment:
We usually add "License" section at the end of each readme (e.g., [here](https://github.com/0xPolygonMiden/miden-vm/blob/main/miden/README.md?plain=1#L279-L280)).